app/src/Book.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace App;
+/**
+ * Book is a DB entity, with a constructor to create itself. The constructor has a default of null and can be passed an individual book id.
+ * @package App
+ */
+class Book
+{
+    public $db;
+    public $id;
+    public $title;
+    public $price;
+    public $description;
+    public $image;
+
+    /**
+     * Book constructor.
+     * @param int $id (book id)
+     * @param PDO $db (database connection)
+     */
+    public function __construct(\PDO $db = null, int $id = null)
+    {
+        $this->db = $db;
+        if (isset($id)) {
+            $query = $this->db->prepare("SELECT `id`, `title`, `price`, `description`, `image` FROM `books` WHERE `id` = :id;");
+            $query->bindParam(":id", $id, \PDO::PARAM_INT);
+            $query->execute();
+            $book = $query->fetch();
+            try{
```

Comment:
no! the try needs to that wrap the database call. setting things on an object cannot throw an exception, but the database call can! wrap the try round everything inside the if(isset...)